src/core/qgserror.h

```diff
@@ -132,7 +132,7 @@ class CORE_EXPORT QgsError
      * \brief messageList return the list of current error messages
      * \return current list of error messages
      */
-    const QList<QgsErrorMessage> messageList() { return mMessageList; }
+    const QList<QgsErrorMessage> messageList() const { return mMessageList; }
```

Comment:
Thanks @elpaso 
Did you review if the const for the return type serves a purpose? I am asking because so far we always asked devs to drop it and it would be nice to have this done consistently throughout the codebase.

Reply:
I was convinced that returning a const here allowed you to range loop iterate on the list without the need to make a const copy of it. But I might be wrong. Let me check if I can get an answer from a debugger.
Btw, I'm now making a const copy as you requested so this const is now unnecessary, but I'm still convinced that if the const copy is not needed we should not make it.

Reply:
I assume even if we don't explicitly make a copy, a temporary for return will be created implicitly anyway, but looking forward to your findings 👍

Reply:
```
    const QList<QgsErrorMessage> errorList( stylesErrors.messageList( ) );
    for ( const auto &err : errorList )
    {
      errors.append( err );
    }

    for ( const auto &err : stylesErrors.messageList( ) )
    {
      errors.append( err );
    }
```
@m-kuhn what I've found is that the instance address or `err` in the two loops is exactly the same when the `messageList` returns a const, but it is different if I remove the const from the signature (the one in front of it).

I think we can conclude that returning a const list allows to range loop iterate without the need to make a const copy of the list.

Btw, I guess you are right about the temporary (should we return a reference?) but to my limited understanding what matters here are the instances in the list, that are not copied, not the container itself.

Reply:
Might be just my understanding:

What that means is, that if you compile with your compiler it actually does make a difference. And this might indeed be backed up by the C++ standard (from C++0x on).

What this also means is, that with the const copy of the list we are always on the safe side, since

> ... what matters here are the instances in the list, that are not copied, not the container itself.

Unless of course we modify all our return values from `QList<X>` to `const QList<X>`, then we will also be on the safe side with the other approach. But only for QGIS internal functions, I think the whole Qt codebase returns non-const lists.

Reply:
no problems to remove the first const if that's the "standard" way for Qt to return lists.

Reply:
I checked with Qt upstream, that's the response:

> Making returned containers const inhibits move semantics, because const rvalues do not bind to the mutable rvalue references that move constructors and move assignment operators use.